src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -27,56 +27,159 @@ internal class SemanticTokensHelpers
+
+                // No specific lsp property for both of these
+                [ClassificationTypeNames.StructName] = SemanticTokenTypes.Struct,
+                [ClassificationTypeNames.RecordStructName] = SemanticTokenTypes.Struct,
+
+                [ClassificationTypeNames.NamespaceName] = SemanticTokenTypes.Namespace,
+                [ClassificationTypeNames.EnumName] = SemanticTokenTypes.Enum,
+                [ClassificationTypeNames.InterfaceName] = SemanticTokenTypes.Interface,
+                [ClassificationTypeNames.TypeParameterName] = SemanticTokenTypes.TypeParameter,
+                [ClassificationTypeNames.ParameterName] = SemanticTokenTypes.Parameter,
+                [ClassificationTypeNames.LocalName] = SemanticTokenTypes.Variable,
+
+                // No specific lsp property for all of these
+                [ClassificationTypeNames.PropertyName] = SemanticTokenTypes.Property,
+                [ClassificationTypeNames.FieldName] = SemanticTokenTypes.Property,
+                [ClassificationTypeNames.ConstantName] = SemanticTokenTypes.Property,
+
+                // No specific lsp property for all of these
+                [ClassificationTypeNames.MethodName] = SemanticTokenTypes.Method,
+                [ClassificationTypeNames.ExtensionMethodName] = SemanticTokenTypes.Method,
+
+                [ClassificationTypeNames.EnumMemberName] = SemanticTokenTypes.EnumMember,
+                [ClassificationTypeNames.EventName] = SemanticTokenTypes.Event,
+                [ClassificationTypeNames.PreprocessorKeyword] = SemanticTokenTypes.Macro,
+
+                // in https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide#standard-token-types-and-modifiers
+                [ClassificationTypeNames.LabelName] = "label",
+
+                // No specific lsp property for all of these
+                [ClassificationTypeNames.RegexComment] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexCharacterClass] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexAnchor] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexQuantifier] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexGrouping] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexAlternation] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexText] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexSelfEscapedCharacter] = SemanticTokenTypes.Regexp,
+                [ClassificationTypeNames.RegexOtherEscape] = SemanticTokenTypes.Regexp,
+
+                // TODO: Missing lsp classifications for xml doc comments, xml literals (vb), json.
+
+                // TODO: Missing specific lsp classifications for the following classification type names.
+
+#if false
+                public const string ExcludedCode = "excluded code";
```

Comment:
We still return these to the client as our custom token type name, correct?

Reply:
correct.